libobs/util/dstr.h

```diff
@@ -287,7 +287,9 @@ static inline const char *dstr_find(const struct dstr *str, const char *find)
 
 static inline int dstr_cmp(const struct dstr *str1, const char *str2)
 {
-	return strcmp(str1->array, str2);
+	const char *s1 = str1->array ? str1->array : "";
+	const char *s2 = str2 ? str2 : "";
+	return strcmp(s1, s2);
```

Comment:
```suggestion
	if (!str1->array)
		return -1;
	if (!str2)
		return 1;
	
	return strcmp(str1->array, str2);
```

Reply:
Is it legal to have a < b, b < a in a cmp? Otherwise seems like a better choice.

Reply:
That's what `strcmp` does with empty strings.

Reply:
empty strings compare equally in strcmp, `strcmp("","") == 0`

Reply:
Ah I see what you mean, yeah that case would be invalid then.